templates/broker-deployment.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: asb
+  namespace: ansible-service-broker
+  labels:
+    app: ansible-service-broker
+    service: asb
+spec:
+  strategy:
+    type: Recreate
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: ansible-service-broker
+        service: asb
+    spec:
+      restartPolicy: Always
+      containers:
+        - image: "ansibleapp/ansible-service-broker-asb:latest"
+          name: main
+          imagePullPolicy: IfNotPresent
+          ports:
+            - containerPort: 1338
+              protocol: TCP
+          env:
+            - name: DOCKERHUB_PASS
+              value: "{{dockerhub_pass}}"
+            - name: DOCKERHUB_USER
+              value: "{{dockerhub_user}}"
+            - name: OPENSHIFT_PASS
+              value: "{{openshift_pass}}"
+            - name: OPENSHIFT_TARGET
+              value: "{{openshift_target}}"
+            - name: OPENSHIFT_USER
+              value: "{{openshift_user}}"
```

Comment:
Because this gets modified can we make this a tmpl file? That is let's rename it to `broker-deployment.yaml.tmpl`. Then when we update the values let's output that into the `broker-deployment.yaml`. Then let's add `broker-deployment.yaml` to .gitignore. 

That way after running make deploy I don't have modified files.

```
$ git status
On branch broker-build-process
Your branch and 'origin/broker-build-process' have diverged,
and have 7 and 4 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   templates/broker-deployment.yaml
```